src/map/utils/synthutils.cpp

```diff
@@ -659,77 +659,80 @@ namespace synthutils
-        {
-            if (slotID != 8)
+            for (uint8 slotID = 1; slotID <= 8; ++slotID)
             {
-                nextSlotID = PChar->CraftContainer->getInvSlotID(slotID + 1);
-            }
-
-            random = xirand::GetRandomNumber(1.);
+                if (slotID != 8)
+                {
+                    nextSlotID = PChar->CraftContainer->getInvSlotID(slotID + 1);
+                }
 
-            if (random < lostItem)
-            {
-                PChar->CraftContainer->setQuantity(slotID, 0);
-                lostCount++;
-            }
-            totalCount++;
+                random = xirand::GetRandomNumber(1.);
 
-            if (invSlotID != nextSlotID)
-            {
-                CItem* PItem = PChar->getStorage(LOC_INVENTORY)->GetItem(invSlotID);
+                if (random < lostItem)
+                {
+                    PChar->CraftContainer->setQuantity(slotID, 0);
+                    lostCount++;
+                }
+                totalCount++;
 
-                if (PItem != nullptr)
+                if (invSlotID != nextSlotID)
                 {
-                    PItem->setSubType(ITEM_UNLOCKED);
-                    PItem->setReserve(PItem->getReserve() - totalCount);
-                    totalCount = 0;
+                    CItem* PItem = PChar->getStorage(LOC_INVENTORY)->GetItem(invSlotID);
 
-                    if (lostCount > 0)
-                    {
-                        charutils::UpdateItem(PChar, LOC_INVENTORY, invSlotID, -(int32)lostCount);
-                        lostCount = 0;
-                    }
-                    else
+                    if (PItem != nullptr)
                     {
-                        PChar->pushPacket(new CInventoryAssignPacket(PItem, INV_NORMAL));
+                        PItem->setSubType(ITEM_UNLOCKED);
+                        PItem->setReserve(PItem->getReserve() - totalCount);
+                        totalCount = 0;
+
+                        if (lostCount > 0)
+                        {
+                            charutils::UpdateItem(PChar, LOC_INVENTORY, invSlotID, -(int32)lostCount);
+                            lostCount = 0;
+                        }
+                        else
+                        {
+                            PChar->pushPacket(new CInventoryAssignPacket(PItem, INV_NORMAL));
+                        }
                     }
+                    invSlotID = nextSlotID;
+                }
+                nextSlotID = 0;
+                if (invSlotID == 0xFF)
+                {
+                    break;
                 }
-                invSlotID = nextSlotID;
-            }
-            nextSlotID = 0;
-            if (invSlotID == 0xFF)
-            {
-                break;
             }
         }
 
-        if (PChar->loc.zone->GetID() != 255 && PChar->loc.zone->GetID() != 0)
+        if (PChar->loc.zone->GetID() != 255 && PChar->loc.zone->GetID() != 0) // Why?
```

Comment:
While you're here, can you swap these out for the enums from zone.h so this is more readable?
This is zone 0: unknown and zone 255: Abyssea-Empyreal_Paradox

and maybe we should think about zones where crafting isn't allowed and expand this list?

Reply:
This check I find it very weird. What it does is just prevent sending the SYNTH_FAIL message to players in range. Thats it.

I will update the ids with their enums, Iwas just wondering the need for this check for those 2 concrete zones.